haiku/_src/data_structures_test.py

```diff
@@ -139,6 +140,13 @@ def test_getattr(self):
     self.assertEqual(f.a, 1)
     self.assertEqual(f.b, 2)
 
+  def test_setattr(self):
+    f = frozendict(a=1, b=2)
+    with pytest.raises(AttributeError, match="can't set attribute"):
```

Comment:
Please use the absltest variant (`with self.assertRaises(E)` or `with self.assertRaisesRegex(E, msg)`) for consistency with the rest of Haiku.

Additionally if you update the impl to use slots you will need to suppress pytype:

```python
def test_setattr(self):
  f = frozendict(a=1)
  with self.assertRaises(AttributeError):
    # Existing attr.
    f.a = 4  # pytype: disable=not-writable
  with self.assertRaises(AttributeError):
    # New attr.
    f.c = 4  # pytype: disable=not-writable
```

Reply:
Thanks, I hadn't used abseil before. Do you have some notes on your typical development setup?

I use sublime with a standard pep8 linter (flake8) and it lights up like a christmas tree.

Reply:
Thanks, I've just realized we totally forgot to document this! Will send a patch tomorrow adding notes to the docs. I'll also add a github action to give warnings/errors on PRs.

For linting we're using `pylint` with the Google [`pylintrc`](https://google.github.io/styleguide/pylintrc) (one amendment is that we also disable `no-name-in-module` because it gets confused by our `del _src` trick at the top level). I'll check our config into Haiku tomorrow so you should be able to just run `pylint haiku` in a checkout of Haiku and it will come back clean.

For Sublime Text I've tested and using the `SublimeLinter` and `SublimeLinter-pylint` extensions (installed with the Sublime package manager) it works (pending the update to Haiku including the pylintrc file that I will push tomorrow). I use pyenv to manage Python and needed to point it to the shims directory to get Sublime working (this is my SublimeLinter config):

```
{
    "paths": {
        "linux": [],
        "osx": [
            "~/.pyenv/shims"
        ],
        "windows": []
    }
}
```